lib/plugins/aws/lib/naming.js

```diff
@@ -284,8 +284,15 @@ module.exports = {
   getMethodLogicalId(resourceId, methodName) {
     return `ApiGatewayMethod${resourceId}${this.normalizeMethodName(methodName)}`;
   },
-  getValidatorLogicalId(modelLogicalId) {
-    return `${modelLogicalId}Validator`;
+  getValidatorLogicalId() {
+    return `ApiGateway${_.upperFirst(
+      this.normalizeNameToAlphaNumericOnly(
+        _.camelCase(
+          // Service name trimmed to 150 chars to avoid going over 255 character limit
+          this.provider.serverless.service.service.slice(0, 150)
```

Comment:
I would remove that `slice`, as anyway we're about to provide restriction of max 20 chars for service name, check https://github.com/serverless/serverless/issues/7056

Reply:
Thanks for the heads up on the service name limit. I have removed the slice.